### PR TITLE
reorganizing readmes

### PR DESCRIPTION
**Pull request by pdg744**
The current readme for the Bonsai foundry template is a huge wall of text. 
Would like to make it feel less scary for devs to start building. 

This PR makes some steps in that direction, by:
- moving the deployment instructions into a dedicated `deployment-guide`
- removing the sections on `Contracts` and `Methods` from the main README 
  - the relevant content has been migrated into `contracts/readme` and `methods/readme` as appropriate

With these changes, the main README is less than half the previous size!

### CODE REVIEW

**Comment by pdg744** [comment]
Note: This PR references some URLs that won't exist until https://github.com/risc0/website/pull/270 lands

**Review by tzerrell — COMMENTED** [review]
I like this, but you should probably also get review from a Bonsai engineer.

**Review by mothran — APPROVED** [review]
This looks like a nice break down to me!

I would tag in @nategraf to for additional review though.

File: README.md [hunk 1]
```diff
@@ -168,57 +68,12 @@ Now if you run `forge test` with `RISC0_DEV_MODE=false`, the test will run as be
...
-    ```bash
-    export BONSAI_API_KEY="YOUR_API_KEY" # see form linked in the previous section
-    export BONSAI_API_URL="BONSAI_URL" # provided with your api key
-    export ALCHEMY_API_KEY="YOUR_ALCHEMY_API_KEY" # the API_KEY provided with an alchemy account
-    export DEPLOYER_PRIVATE_KEY="YOUR_WALLET_PRIVATE_KEY" # the private key of your Ethereum testnet wallet e.g., Sepolia
-    ```
-
-2.  Deploy an `IBonsaiRelay` contract by running:
+## Next Steps
+To build a basic Bonsai project, you'll need to make changes in two folders:
```
Comment (nategraf) [inline README.md:72]: Suggest "your application" to indicate that this is how to achieve their goal
```suggestion
To build your application, you'll need to make changes in two folders: 
```

File: README.md [hunk 2]
```diff
@@ -248,27 +103,10 @@ Below are the primary files in the project directory
...
-
-Code in the `methods/guest` directory will be compiled into one or more [RISC-V] binaries.
-Each will have a corresponding image ID, which is a hash identifying the program.
-When deploying your application, you will upload your binary to Bonsai where the guest will run when requested.
-The image ID will be included in the deployment of the smart contracts to reference your guest program living in Bonsai.
-
-Build configuration for the methods is included in `methods/build.rs`.
 
+[methods]: /methods/README
+[contracts]: /contracts/README
```
Comment (nategraf) [inline README.md:108]: These links are broken
```suggestion
[methods]: /methods
[contracts]: /contracts
```

**Review by nategraf — APPROVED** [review]
Approved, with a could of minor comments

**Pull request merged**